### ISSUE TITLE
autocompletion tricky to use - needs <space> followed by <backspace> to work

### ISSUE THREAD

**Issue by benintech** [carlitux/deoplete-ternjs#90]
Hi,
My tests have been done first with Vim 8.1 (installed from Debian 10 package), then with the latest Vim (8.2), installed from source. Same result. The plugins deoplete and deoplete-ternjs are up to date, so is Node and Tern.

vimrc is empty, I disabled all plugins and extra configurations. For deoplete-ternjs's configuration I copied the example of your main Github page.

It works, but it's cumbersome. Please take a look at the 30 seconds screencast below.

Let's say I have an array, myNumbers. If I tape myNumbers followed by a dot, I expect all array methods to pop up in the completion list. It happens indeed, but only after some tricky additional steps  :

- If I tape myNumbers followed by dot, nothing happens. I have to tape an additional space to see the completion list, with indeed the array methods. Then backspace (to go back to the dot), the method list is preserved
- Let's say I then select the splice() method.
- If I tape myNumbers followed by dot again, it gets worse. Only the splice() method pops up, from the buffer source (and not Tern). Again, I have to tape a space to see all methods for an array. But if I then tape backspace (to go back to the dot), again only the splice() method is proposed. I have then to delete the dot, the re-tape the dot, and then a lot of TernJS functions are proposed, but not the array methods. Then I tape space again, followed by backspace, and (at least) I get all the methods of an array.
So, basically I have to tape dot, space, backspace, backspace, dot again, space, backspace, and then I will (at last) get my expected result.

I tried to modify the min_pattern_length for tern source, but without result (I tried `call deoplete#custom#source('tern', 'min_pattern_length', 0)` and `call deoplete#custom#source('tern', 'min_pattern_length', 1)`)

Is this really supposed to work like this ? Am I missing something ? Is deoplete-ternjs only supposed to work with NeoVim, and not legacy Vim ?

https://user-images.githubusercontent.com/5140169/107209759-87c1c000-6a03-11eb-9848-53731b5e3659.mp4

**Comment by carlitux** [comment]
@benintech I can see there is an error when you type dot character so I think there is the issue but I can take a look right now because I have no time and I migrated to the "native" LSP neovim feature. If you can make a PR would be great.

**Comment by carlitux** [comment]
maybe this could help https://github.com/carlitux/deoplete-ternjs/issues/88#issuecomment-713913170

**Comment by benintech** [comment]
> maybe this could help [#88 (comment)](https://github.com/carlitux/deoplete-ternjs/issues/88#issuecomment-713913170)

YES !!! It did the trick !
Everything works perfectly now
THANKS !
